zero_bin/rpc/src/main.rs

```diff
@@ -56,6 +56,13 @@ impl Cli {
             } => {
                 let checkpoint_block_number =
                     checkpoint_block_number.unwrap_or((start_block - 1).into());
+                if checkpoint_block_number != (start_block - 1).into() {
```

Comment:
I would put all checkpoint related checks in one function. You have already made it ` check_previous_proof`. It is easier then to reason about it. You just call it from `client_main`, and return error from it if some of the checks fails.